apidump/apidump.go

```diff
@@ -399,6 +338,30 @@ func Run(args Args) error {
 	// Backend collectors that need trace rotation
 	var toRotate []trace.LearnSessionCollector
 
+	dumpSummary := NewSummary(
+		capturingNegation,
+		interfaces,
+		negationFilters,
+		numUserFilters,
+		filterSummary,
+		prefilterSummary,
+		negationSummary,
```

Comment:
These summaries are read at the same time as they are being updated by the live packet capture. Without synchronization between the two threads, there is no guarantee that the reader will see any writes.

Reply:
Under the hood, these summaries use a `sync.RWMutex`.  The reader can't acquire a new read lock if there is a pending write lock, which should ensure that any pending writes preceding the stats dump are resolved before the stats dump can proceed.  Is there something there I'm missing?

Reply:
Ah, you're right. Sorry I missed this!